libs/cardano-ledger-binary/src/Cardano/Ledger/Binary/Decoding/DecCBOR.hs

```diff
@@ -404,31 +451,21 @@ instance DecCBOR UTCTime where
 
 instance DSIGNAlgorithm v => DecCBOR (VerKeyDSIGN v) where
   decCBOR = decodeVerKeyDSIGN
+  {-# INLINE decCBOR #-}
 
 instance DSIGNAlgorithm v => DecCBOR (SignKeyDSIGN v) where
   decCBOR = decodeSignKeyDSIGN
+  {-# INLINE decCBOR #-}
 
 instance DSIGNAlgorithm v => DecCBOR (SigDSIGN v) where
   decCBOR = decodeSigDSIGN
+  {-# INLINE decCBOR #-}
 
 --------------------------------------------------------------------------------
 -- Hash
 --------------------------------------------------------------------------------
 
-instance (HashAlgorithm h, Typeable a) => DecCBOR (Hash h a) where
-  decCBOR = do
-    bs <- decodeBytes
-    case hashFromBytes bs of
-      Just x -> return x
-      Nothing ->
-        fail $
-          "hash bytes wrong size, expected "
-            ++ show expected
-            ++ " but got "
-            ++ show actual
-        where
-          expected = sizeHash (Proxy :: Proxy h)
-          actual = BS.length bs
+instance (HashAlgorithm h, Typeable a) => DecCBOR (Hash h a)
```

Comment:
whoa, how does this work (removing `decCBOR`)?

Reply:
Default implementation for `DecCBOR` falls back onto `FromCBOR`. In case of Hash they were identical.

Reply:
ah, so I guess the size is checked in `FromCBOR`, thanks